src/spec/dsl.cr

```diff
@@ -110,6 +110,18 @@ module Spec
     lines << line
   end
 
+  # :nodoc:
+  def self.add_tag(tag)
+    set =
+      if anti_tag = tag.lchop? '~'
+        tag = anti_tag
+        @@anti_tags ||= Set(String).new
+      else
+        @@tags ||= Set(String).new
+      end
+    set << tag
```

Comment:
```cr
if anti_tag = tag.lchop('~')
  (@@anti_tags ||= Set(String).new) << anti_tag
else
  (@@tags ||= Set(String).new) << tag
end
```

Bonus points for just using `class_getter tags = Set(String).new`

Reply:
@RX14 Shouldn't that be `class_getter tags : Set(String) { Set(String).new }` (the lazy variant)?

Reply:
Why does it matter? Whats the benefit of shaving a new ns of the class initialization time here?

Reply:
The code as is follows the pattern of the other filters with respect to how it sets an @@ var, so while it may make it cleaner to change it, I think that's best for a follow up PR.

I don't think we want the class_getter with a default because a nil `@@tags` (or `@@locations` or any other filter) actually means that it wasn't passed on the cli, and that knowledge is used later.  I could see changing that, and relying on, say, checking for an empty set, but as I said that's definitely for a follow up PR